.ado/templates/prepare-env.yml

```diff
@@ -54,7 +54,7 @@ steps:
         -InstallerUri ${{ parameters.vsInstallerUri }}
         -Components ${{ parameters.vsComponents }}
         -Collect:$${{ parameters.debug }}
-    condition: and(ne('${{parameters.vsComponents}}', ''), eq(variables['VmImage'], 'windows-2019'))
+    condition: and(ne('${{ parameters.vsComponents }}', ''), eq(variables['VmImage'], 'windows-2019')) # Note that this is evaluated at parsing, so parameters.vsComponents is never empty
```

Comment:
If I'm understanding right, why not remove that part of the condition if it will always be true? (same in other file)

Reply:
We need some condition such that if vsComponents resolves to '', we don't run this step. Condition is evaluated at parse time, comparing `'($VsComponents)' != '')`. Even if VsComponents is empty, the variable hasn't been resolved yet.

The solution is to put the parameter back into a variable, and compare that with the variables syntax. Unfortunately, we can't define new variables down in the template. I think figuring this out is out of scope for this PR, since it turns out, even if we stop installing the v141 toolset, the windows-2019 VMs are actually also missing most of the UWP C++ toolset anyway, so we still need this task to run.

So rather than change anything here, I added the comment in case in the future we actually get VMs already properly set up with VS 2019.